geode-web-api/src/main/java/org/apache/geode/rest/internal/web/controllers/PdxBasedCrudController.java

```diff
@@ -295,12 +296,13 @@ protected String getRestApiVersion() {
       @ApiResponse(code = 500, message = "GemFire throws an error or exception.")})
   @PreAuthorize("@securityService.authorize('WRITE', #region, #keys)")
   public ResponseEntity<?> update(@PathVariable("region") String region,
-      @PathVariable("keys") final String[] keys,
+      @PathVariable("keys") String[] keys,
```

Comment:
Have you considered just using the Spring annotations and patterns to match the remainder of the URL path?

`@RequestMapping(value = "/{region}/**", ...)`
and
`String[] keys = request.getAttribute(HandlerMapping.PATH_WITHIN_HANDLER_MAPPING_ATTRIBUTE).split(",")`

Reply:
It is not clear to me why using /** would be an better than the current solution. The current code is allowing spring to parse the pathvar into an array of String. Why is your way of doing it explicitly better? And how does this relate to supporting a slash in a key? Would your solution not require the keys to be encoded by the user? This /** solution is discouraged here: https://www.baeldung.com/spring-slash-character-in-url but so is encoding/decoding. The current code base already did decoding of each {region} pathvar so that is why I went with also decoding the {keys} pathvar. Also if we don't decode then the swagger page will not work with keys because swagger automatically encodes path and query vars when you execute the apis from the web page.

Reply:
It relates to supporting slashes in the path without needing to encode them. It makes sense to do it they way you proposed if we are already expecting the region to be encoded.